### PR TITLE
Auto add Admin prefix only for Admin nodes.

### LINKED ISSUE #13943: Auto add Admin prefix only for Admin nodes.
We recently updated the `NavigationManager` so that admin menu items are auto prefixed by the `AdminUrlPrefix` option if not yet done, but as stated by @Piedone https://github.com/OrchardCMS/OrchardCore/pull/12807#issuecomment-1616928666, currently the auto admin prefixing should only be done for menu items built by the `LinkAdminNodeNavigationBuilder`.

This to not break people using the `NavigationManager` with other menu nodes but not for admin menus.

### PR DESCRIPTION
**Pull request by jtkech**
Fixes #13943 

We recently updated the `NavigationManager` so that admin menu items are auto prefixed by the `AdminUrlPrefix` option if not yet done, but as stated by @Piedone https://github.com/OrchardCMS/OrchardCore/pull/12807#issuecomment-1616928666, currently the auto admin prefixing should only be done for menu items built by the `LinkAdminNodeNavigationBuilder`.

This to not break people using the `NavigationManager` with other menu nodes but not for admin menus.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs:1]: Why not fully revert the changes of https://github.com/OrchardCMS/OrchardCore/pull/12807/files#diff-09302c08752ef136263675f9d9bc818bbd2f2a5fcaa00eb7a684c738ef82dfd2 though? And the compartmentalize what's needed for the auto-prefixing in `LinkAdminNodeNavigationBuilder`?

**Comment by Skrypt** [comment]
@hishamco NavigationManager is used elsewhere than the admin UI. I use it to build frontend dynamic menus.

File: src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs
Comment (jtkech) [reply src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs:1]: This is what I first did, only auto-prefixing in `LinkAdminNodeNavigationBuilder` but then the url is persisted with the prepended admin prefix, that"s okay because when the prefix is there it is not re-added again, but then if you change the `AdminUrlPrefix` option you will get an url starting with the new prefix but still having the old prefix `newPrefix/oldPrefix/relativePath`.

**Comment by Piedone** [comment]
We do that too with [Lombiq Base Theme](https://github.com/Lombiq/Orchard-Base-Theme).

File: src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs
Comment (Piedone) [reply src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs:1]: I mean, I wouldn't prefix the URLs for storage, just when the menu items are read from the DB and added to the navigation builder.

Comment (jtkech) [reply src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs:1]: Yes, this was my first intention, I will re-try it

Comment (jtkech) [reply src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs:1]: Okay, looks like it is possible, maybe I was mutating the `node.LinkUrl` ;)

I will update it

Comment (Piedone) [reply src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs:1]: Thank you!

**Comment by hishamco** [comment]
> @hishamco NavigationManager is used elsewhere than just the admin UI. I use it to build frontend dynamic menus.

Did I ask a question or write a comment about this?

**Comment by Skrypt** [comment]
You just did ask a question :smile: 
Not sure what you mean there.

File: src/OrchardCore.Modules/OrchardCore.Lists/AdminNodes/ListsAdminNodeNavigationBuilder.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class ListsAdminNodeNavigationBuilder : IAdminNodeNavigationBuilder
         private ListsAdminNode _node;
         private ContentTypeDefinition _contentType;
 
-        private const int MaxItemsInNode = 100; // security check
+        private const int _maxItemsInNode = 100; // security check.
```
[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Lists/AdminNodes/ListsAdminNodeNavigationBuilder.cs:28]: Yes I agree, but Visual Studio shows a warning, because of the new rule we have on private fields (leading underscore required) in the editor config, maybe there is a way to exclude `private const` from this rule. Note: We also have warnings on private localizer fields as `S` and `H`.

[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Lists/AdminNodes/ListsAdminNodeNavigationBuilder.cs]: Okay I reverted the renaming of this const

I suggested here #13953 to ignore the underscore rule for private const.

File: src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/LinkAdminNodeNavigationBuilder.cs [hunk 2]
```diff
@@ -67,20 +90,5 @@ public Task BuildNavigationAsync(MenuItem menuItem, NavigationBuilder builder, I
                 }
             });
         }
-
-        // Add adminNode's IconClass property values to menuItem.Classes.
-        // Add them with a prefix so that later the shape template can extract them to use them on a <i> tag.
-        private void AddIconPickerClassToLink(string iconClass, NavigationItemBuilder itemBuilder)
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/LinkAdminNodeNavigationBuilder.cs:73]: While this method is indeed not used, its referenced here too: https://github.com/OrchardCMS/OrchardCore/blob/8cc817c13d12ec39f4dbdec6f9c71195c0c511c1/src/docs/reference/modules/AdminMenu/README.md?plain=1#L142

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/LinkAdminNodeNavigationBuilder.cs (hunk 2 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/LinkAdminNodeNavigationBuilder.cs:73]: Yes the doc is obsolete, I assume that you meant to just update the doc?

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/LinkAdminNodeNavigationBuilder.cs:73]: If the method doesn't make sense then yes.

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/LinkAdminNodeNavigationBuilder.cs:73]: Okay done

**Review by Piedone — APPROVED** [review]
